sycl/doc/syclcompat/README.md

```diff
@@ -1729,6 +1729,140 @@ static void invoke_kernel_function(kernel_function &function,
 
 } // namespace syclcompat
 ```
+The kernel_launcher class provides a utility for launching SYCL kernels
+through function wrappers, managing execution configurations such as SYCL
+queues, nd_range, and local memory size as thread-local variables. It allows
+kernel invocation through multiple overloads of launch, either by directly
+calling a function pointer with arguments, invoking a registered function
+wrapper, or using packed argument arrays for flexible invocation. The class
+also maintains a mapping of raw function pointers to launchable functors,
+enabling dynamic function registration.
+
+The wrapper_register class serves as a helper for registering kernel functions
+and invoking them through a wrapper. It supports constructing from a kernel
+function pointer,overloading operator() for kernel execution, and retrieving
+the original function pointer via get(), providing a mechanism for managing
+and invoking SYCL kernels dynamically.
```

Comment:
What does

> providing a mechanism for managing and invoking SYCL kernels dynamically.

mean? Is that the intention of the class as a whole? If so, this should probably be a separate sentence with a little more explanation.